### PR TITLE
Remove Deprecated XContentHelper.createParser Call in DocumentParser

### PR DESCRIPTION
**Pull request by original-brownbear**
Remove the use of the deprecated xcontent helper method because it triggered
xcontent type guessing needlessly.
Also, fixed the deprecated method from not needlessly guessing content type (only realized it's this easy to fixing the above ... but code is cleaner now regardless).

### CODE REVIEW

**Review by romseygeek — APPROVED** [review]
Nice cleanup, thanks @original-brownbear

File: server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java [hunk 1]
```diff
@@ -86,7 +86,11 @@ public static XContentParser createParser(
         BytesReference bytes,
         XContentType xContentType
     ) throws IOException {
-        return createParser(XContentParserConfiguration.EMPTY.withRegistry(registry).withDeprecationHandler(deprecation), bytes);
+        return createParser(
+            XContentParserConfiguration.EMPTY.withRegistry(registry).withDeprecationHandler(deprecation),
+            bytes,
+            xContentType
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java:92]: arg good catch!

Comment (javanna) [reply server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java:92]: given that you fixed content type auto guessing here, I wonder if the switch from registry to configuration is still needed. I must admit I am not familiar with the configuration though so I may totally be missing something...

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java:92]: I guess it's not technically "needed" but it moves us away from a deprecated API and saves some (though probably irrelevant) objects by not having to set up the config on every call.

**Comment by original-brownbear** [comment]
Thanks Alan + Luca!

**Pull request merged**